tools/Dashboard/DevHome.Dashboard/Helpers/WidgetIconCache.cs

```diff
@@ -24,28 +25,29 @@ internal class WidgetIconCache
     /// <remarks>
     /// The WidgetIconCache is backed by two dictionaries, one for light themed icons and one for dark themed icons.
     /// </remarks>
-    public WidgetIconCache()
+    public WidgetIconCache(DispatcherQueue dispatcher)
     {
+        _dispatcher = dispatcher;
         _widgetLightIconCache = new Dictionary<string, BitmapImage>();
         _widgetDarkIconCache = new Dictionary<string, BitmapImage>();
     }
 
     /// <summary>
     /// Caches icons for all widgets in the WidgetCatalog that are included in Dev Home.
     /// </summary>
-    public async Task CacheAllWidgetIcons(WidgetCatalog widgetCatalog, DispatcherQueue dispatcher)
+    public async Task CacheAllWidgetIconsAsync(WidgetCatalog widgetCatalog)
     {
         var widgetDefs = widgetCatalog.GetWidgetDefinitions();
         foreach (var widgetDef in widgetDefs ?? Array.Empty<WidgetDefinition>())
         {
-            await AddIconsToCache(widgetDef, dispatcher);
+            await AddIconsToCacheAsync(widgetDef);

```

Comment:
Won't this method run synchronously?  Because the foreach isn't awaited `CacheAllWidgetsAsync` waits for the foreach loop to finish before giving control back to the caller.

Reply:
Good catch. Updated.